### ISSUE TITLE
Questions about equation (1) and (3)

### ISSUE THREAD

**Issue by YeRen123455** [jbeomlee93/AdvCAM#18]
Hi @jbeomlee93  I have some questions for the equations of you paper. That is:
    1. For equation(1), the paper[16] uses x' = x + gradient  to adversarial attack the input image (i.e., X) and finally generate the image Xadv, which is opposited to your equation (1) (i.e., x' = x - gradient ).
    2. For equation(3),  the paper[28] uses  iter(x' = x + gradient) to represents the adversarial attack process, which is same to equation (3)(i.e., x' = x + gradient ). I think adversarial climbing should be opposited to the adversarial attack. 
   Sincerely waiting for your reply !

**Comment by jbeomlee93** [comment]
Hi @YeRen123455,

Sorry for the confusion. [16, 28] manipulate images with perturbation obtained by "Loss", but our method manipulates with "Logit".

Since decreasing the logit value increases the loss value, Eq (1) and [16] are conceptually the same.

Thanks

**Comment by YeRen123455** [comment]
@jbeomlee93 
Thanks a lot!

**Comment by YeRen123455** [comment]
@jbeomlee93   Thanks a lot! I get it.
Since the image label is available, why don't you use "LOSS" to manipulate the images.

**Comment by jbeomlee93** [comment]
We perform adversarial climbing on training images, and the trained model already produces very small loss values for the training images. As such, loss-based adversarial climbing rarely manipulates images.

**Comment by YeRen123455** [comment]
@jbeomlee93  Thanks for your answer!  Sorry for disturbing you again. I still have two questions about the released code.
(1) In obtain_CAM_masking_super_pixel.py.  Since you have used grad-cam to generate the class activation map(i.e., CAM), why don't you use resnet50.py with grad-cam to generate outputs. Actually, you used resnet50_cam.py with grad-cam to generate the outputs.

(2) Can you share the code of  "SEAM+AdvCAM" with me. I try to reproduce it by myself but the performance is not good as yours.   My email  address is liboyang20@nudt.edu.cn
